### PR TITLE
Upgrade to Quarkus 2.12.0.CR1

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -277,6 +277,10 @@
                                             <project.groupId>${project.groupId}</project.groupId>
                                         </systemProperties>
                                     </defaultTestConfig>
+                                    <!--
+                                    Build step 'org.kie.kogito.infinispan.quarkus.deployment.KogitoAddOnPersistenceInfinispanProcessor#verifyCapabilities' does not produce any build item and thus will never get executed
+                                    -->
```
Comment (gsmet) [inline pom.xml:282]: @evacchi @cristianonicolai @mbiarnes this commit will have to be reverted when we upgrade Kogito.

Comment (evacchi) [reply pom.xml:282]: /cc @radtriste

Comment (radtriste) [reply pom.xml:282]: oki
cc @mbiarnes

Comment (radtriste) [reply pom.xml:282]: This would be for Kogito 1.27 then

**Pull request merged**